app/controllers/spots_controller.rb

```diff
@@ -30,12 +30,18 @@ def edit
 
   def update
     @spot = Spot.find(params[:id])
-    if @spot.update(spot_params)
-      flash[:success] = "Spot updated."
-      redirect_to spot_path(@spot)
+    if (current_user && current_user.id == @spot.user_id) ||
+      (current_user && current_user.admin?)
```

Comment:
Use 4 (not 2) spaces for indenting a condition in an `if` statement spanning multiple lines.